### PR TITLE
feat: add native Dockerfile to create ansible free image

### PR DESCRIPTION
**Pull request by iamsobanjaved**
Details in issue: https://github.com/edx/edx-arch-experiments/issues/276

### CODE REVIEW

File: .github/workflows/docker-publish.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        with:
+          script: |
+            const branchName = context.ref.split('/').slice(-1)[0];
+            const tagName = branchName === 'master' ? 'latest' : branchName;
+            console.log('Will use tag: ' + tagName);
+            return tagName;
+          result-encoding: string
+
+      - name: Build and push Dev Docker image
+        uses: docker/build-push-action@v1
```
Comment (UsamaSadiq) [inline .github/workflows/docker-publish.yml:31]:
```suggestion
        uses: docker/build-push-action@v4
```

File: .github/workflows/docker-publish.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      # suitable images for smaller prod environments.
+      # - name: Build and push prod Docker image
+      #   uses: docker/build-push-action@v1
+      #   with:
+      #     push: true
+      #     username: ${{ secrets.DOCKERHUB_USERNAME }}
+      #     password: ${{ secrets.DOCKERHUB_PASSWORD }}
+      #     target: prod
+      #     repository: edxops/ecommerce-prod
+      #     tags: ${{ steps.get-tag-name.outputs.result }},${{ github.sha }}
```
Comment (UsamaSadiq) [inline .github/workflows/docker-publish.yml]:
```suggestion
      #     tags: ${{ steps.get-tag-name.outputs.result }},${{ github.sha }}

```

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+# Install JS requirements
+COPY package.json package.json
+COPY package-lock.json package-lock.json
+COPY bower.json bower.json
+RUN npm install --production && ./node_modules/.bin/bower install --allow-root --production
+
+# Expose canonical ecommerce port
+EXPOSE 18130
+EXPOSE 8130
```
Comment (UsamaSadiq) [inline Dockerfile]: Is this port `8130` also needed?

Comment (UsamaSadiq) [inline Dockerfile]:
```suggestion
```

File: .github/workflows/docker-publish.yml (hunk 1 above)
Comment (iamsobanjaved) [reply .github/workflows/docker-publish.yml:31]: This has some issues, will create a ticket for upgrading this action

File: Dockerfile (hunk 3 above)
Comment (iamsobanjaved) [reply Dockerfile]: removed

**Review by UsamaSadiq — APPROVED** [review]

**Review by pshiu — APPROVED** [review]

**Comment by iamsobanjaved** [comment]
@pshiu I don't have merge access, can you merge this so I can merge related devstack PR?

**Pull request merged**